_drafts/2015-10-02-microcoap-and-ff-copper.md

```diff
@@ -0,0 +1,245 @@
+---
+layout: post
+title: CoAP on RIOT
+subtitle: implementing and testing with microcoap and copper
+categories: CoAP
+author: lotte
+date: 2015-03-17
+---
+
+In [one of our last posts](http://watr.li/what-is-coap.html), we explained how the [Constrained Application Protocol](http://coap.technology) enables us to exchange data between nodes in the Internet of Plants using a request/reply cycle similar to that of HTTP. 
+To do this, we need both of our node types– plant nodes and display nodes– to be able to send, process and answer CoAP requests.
+
+This post explains how to implement a simple CoAP server on our *plant nodes*, which run [RIOT](http://www.riot-os.org), using RIOTs [microcoap](https://github.com/1248/microcoap) package. 
+
+Since no special modifications to the code are needed to get it to run on RIOT, this guide may also be useful to you if you're looking to run microcoap on Linux or your Arduino.{: .alert .alert-info }
+
+We'll also show you how to test your microcoap server with [Copper](https://addons.mozilla.org/de/firefox/addon/copper-270430/), using [marz](https://github.com/sgso/marz) to tunnel the requests to your RIOT instance.
+
+## Implementing a simple microcoap server
+This section will walk you through the implementation of a very simple microcoap server. In the end, you'll have a server which is able to answer ``GET`` requests to ``/foo/bar``, and (hopefully :) ) the knowledge how to extend this server at will. This guide is based on the code of [this example application](https://github.com/RIOT-OS/applications/tree/master/microcoap). It is advised that you keep an eye on it as a reference. Let's go through the code file by file.
```

Comment:
passive voice ("it is advised") klingt hier ein wenig hochgestochen finde ich :D